Assets/Scripts/Controllers/ModsManager.cs

```diff
@@ -47,27 +49,59 @@ public DirectoryInfo[] GetMods()
     }
 
     /// <summary>
-    /// Loads the base and the mods scripts.
+    /// Loads all the functions from the given script.
+    /// </summary>
+    /// <param name="functionsName">The functions name.</param>
+    /// <param name="fileName">The file name.</param>
+    private void LoadFunctions(string functionsName, string fileName)
+    {
+        LoadTextFile(
+            "LUA",
+            fileName,
+            (filePath) =>
+            {
+                string text = File.ReadAllText(filePath);
+                FunctionsManager.Get(functionsName).LoadScript(text, functionsName);
+            });
+    }
+
+    /// <summary>
+    /// Loads all the character names from the given file.
     /// </summary>
-    /// <param name="mods">The mods directories.</param>
     /// <param name="fileName">The file name.</param>
-    private void LoadFunctionScripts(string functionsName, string fileName)
+    private void LoadCharacterNames(string fileName)
     {
-        string filePath = Path.Combine(Application.streamingAssetsPath, "LUA");
+        LoadTextFile(
+            "Data",
+            fileName,
+            (filePath) =>
+            {
+                string[] lines = File.ReadAllLines(filePath);
+                CharacterNameManager.LoadNames(lines);
+            });
+    }
+
+    /// <summary>
+    /// Loads the given file from the given folder in the base and inside the mods and calls the Delegate.
```

Comment:
Would say something like "...and calls the delegate with the text of each file."